app/assets/javascripts/graphiql/rails/graphiql_show.js

```diff
@@ -67,27 +67,32 @@ document.addEventListener("DOMContentLoaded", function(event) {
     })
   }
 
-  var initial_query = graphiqlContainer.dataset.initialQuery;
+  function GraphiQLInitializer() {
+    const [query, setQuery] = React.useState(graphiqlContainer.dataset.initialQuery || undefined)
+    
+    const enabledPlugins = [];
+    
+    if (graphiqlContainer.dataset.explorerPluginEnabled === 'true') {
+      const explorerPlugin = GraphiQLPluginExplorer.useExplorerPlugin({
+        query,
+        onEdit: setQuery,
+      });
 
-  if (initial_query) {
-    var defaultQuery = initial_query;
-  } else {
-    var defaultQuery = undefined;
-  }
+      enabledPlugins.push(explorerPlugin);
+    }
 
+    var elementProps = { fetcher: graphQLFetcher, 
+      query, 
+      onEditQuery: setQuery, 
+      defaultEditorToolsVisibility: true,
+      plugins: enabledPlugins
+    }
 
-  // Render <GraphiQL /> into the body.
-  var elementProps = { fetcher: graphQLFetcher, defaultQuery: defaultQuery, headerEditorEnabled: graphiqlContainer.dataset.headerEditorEnabled === 'true' };
-  
-  Object.assign(elementProps, { query: parameters.query, variables: parameters.variables })
-  if (queryParams === 'true') {
-    Object.assign(elementProps, { onEditQuery: onEditQuery, onEditVariables: onEditVariables });
+    return React.createElement(GraphiQL, elementProps);
   }
 
   ReactDOM.render(
```

Comment:
React DOM's documentation mentions that the usage of the library in the client is
```js
const root = createRoot(document.getElementById('root'));
root.render(<App />);
```

Did you consider doing that?